### PR TITLE
CClient: get_account_data refactoring

### PR DESCRIPTION
**Pull request by oopsmonk**
* Storing balance values from addresses.
* Adding `account_data_init` and `account_data_clear` for UX. 
* remove `cclient/api/extended/types.h`

### CODE REVIEW

File: cclient/api/extended/get_new_address.h [hunk 1]
```diff
@@ -17,13 +17,22 @@
...
 
-#include "cclient/api/extended/types.h"
 #include "cclient/http/http.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+/**
+ * @brief Address genrating options
```
Comment (thibault-martinez) [inline cclient/api/extended/get_new_address.h]:
```suggestion
 * @brief Address generation options
```

File: cclient/api/extended/get_new_address.h [hunk 2]
```diff
@@ -17,13 +17,22 @@
...
+ *
+ */
+typedef struct {
+  uint64_t start;   /*!< start key index */
+  uint64_t total;   /*!< total key index */
+  uint8_t security; /*!< security level */
+} address_opt_t;
+
 /**
  * @brief Generates and retruns addresses including an unused address.
```
Comment (thibault-martinez) [inline cclient/api/extended/get_new_address.h]:
```suggestion
  * @brief Generates and returns addresses including an unused address.
```

File: cclient/api/extended/get_account_data.h [hunk 3]
```diff
@@ -17,13 +17,46 @@
...
+
+/**
+ * @brief Clean up the #account_data_t.
+ *
+ * @param[out] account An #account_data_t object.
+ */
+void account_data_clear(account_data_t* const account);
+
+/**
+ * @brief
```
Comment (thibault-martinez) [inline cclient/api/extended/get_account_data.h]: No brief ?

File: cclient/api/extended/get_account_data.h [hunk 4]
```diff
@@ -17,13 +17,46 @@
...
+ *
+ * @param[in] account Gets the balance value by index.
```
Comment (thibault-martinez) [inline cclient/api/extended/get_account_data.h]: Missing a parameter

File: cclient/api/extended/get_account_data.c [hunk 5]
```diff
@@ -4,22 +4,43 @@
...
+void account_data_init(account_data_t* const account) {
+  account->balance = 0;
+  memset(account->latest_address, 0, FLEX_TRIT_SIZE_243);
+  account->addresses = NULL;
+  account->transactions = NULL;
+  account->balances = NULL;
+  utarray_new(account->balances, &ut_balance_icd);
+}
+
+void account_data_clear(account_data_t* const account) {
```
Comment (thibault-martinez) [inline cclient/api/extended/get_account_data.c:28]: Why not `account_data_free` ? It doesn't really "clear"

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

File: cclient/api/extended/get_account_data.c (hunk 5 above)
Comment (oopsmonk) [reply cclient/api/extended/get_account_data.c:28]: it's not exactly freeing the `account_data_t` object.

Comment (thibault-martinez) [reply cclient/api/extended/get_account_data.c:28]: Alright!

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**